numpy/random/mtrand/mtrand.pyx

```diff
@@ -936,37 +936,14 @@ cdef class RandomState:
                [3, 2, 2, 0]])
 
         """
-        cdef long lo, hi, rv
-        cdef unsigned long diff
-        cdef long *array_data
-        cdef ndarray array "arrayObject"
-        cdef npy_intp length
-        cdef npy_intp i
+        if high is not None and low >= high:
+            raise ValueError("low >= high")
 
         if high is None:
-            lo = 0
-            hi = low
-        else:
-            lo = low
-            hi = high
-
-        if lo >= hi :
-            raise ValueError("low >= high")
+            high = low
+            low = 0
 
-        diff = <unsigned long>hi - <unsigned long>lo - 1UL
-        if size is None:
-            with self.lock:
-                rv = lo + <long>rk_interval(diff, self. internal_state)
-            return rv
-        else:
-            array = <ndarray>np.empty(size, int)
-            length = PyArray_SIZE(array)
-            array_data = <long *>PyArray_DATA(array)
-            with self.lock, nogil:
-                for i from 0 <= i < length:
-                    rv = lo + <long>rk_interval(diff, self. internal_state)
-                    array_data[i] = rv
-            return array
+        return self.random_integers(low, high-1, size)
```

Comment:
Sure thing.

By the way, in relation to the discussion in #6869, where does this PR stand in terms of being merge-able?  Can the changes go in as is (I don't forsee #6824 or #6869 being merged in for some time at this point given what has transpired)?  Or would additional work be needed?